### ISSUE TITLE
Menu - Menu items announced with incorrect state information

### ISSUE THREAD

**Issue by catherinehynes** [Esri/calcite-components#13775]
**monday.com sync:** #11102718019

### Check existing issues

- [x] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates and reviewed the [Accessibility page](https://developers.arcgis.com/calcite-design-system/foundations/accessibility) for guidance.

### Which Component

Menu

### Summary

An internal central accessibility team [test in ArcGIS Enterprise 12.0](https://devtopia.esri.com/WebGIS/arcgis-portal-app/issues/63278) found that the menu provides incorrect state information to screen readers, announcing expanded items as collapsed.

### Actual Behavior

With a screen reader (NVDA) running, navigate through the menu and note that the announcement for all items includes "collapsed" state information: 

<img width="1916" height="752" alt="Image" src="https://github.com/user-attachments/assets/eb074d73-673e-4a4b-9468-36ed19ece658" />

### Expected Behavior

The currently expanded menu item should be announced as expanded, not collapsed.

### Reproduction Sample

https://codepen.io/chynes27/pen/zxBPzPb?editors=100

### Reproduction Steps

1. Open sample
2. Start NVDA
3. Use keyboard to navigate through the menu of Drivers, Routes, and Forecast
4. Note that NVDA announces each menu item as collapsed, even Routes, which is the current item and should be expanded

### Reproduction Version

3.3.3

### Relevant Info

Tested with NVDA on Chrome

### Priority impact

impact - p2 - want for an upcoming milestone

### Calcite package

- [x] @esri/calcite-components
- [ ] @esri/calcite-components-react
- [ ] @esri/calcite-design-tokens
- [ ] @esri/eslint-plugin-calcite-components

### Esri team

ArcGIS Enterprise

**Comment by macandcheese** [comment]
Might be good to dig deeper here and see if `active` is the state here that should be communicated to the user. We might also be able to conditionally disregard collapsed / expanded altogether, really that should be related to the collapse or expanded state of nested and child Menu Items, not page content outside the component.
